library/ripemd160.c

```diff
@@ -460,6 +495,12 @@ int mbedtls_ripemd160_self_test( int verbose )
         mbedtls_printf( "\n" );
 
     return( 0 );
+
+fail:
+    if( verbose != 0 )
+        mbedtls_printf( "failed\n" );
+
+    return( ret );
 }
```

Comment:
I do not know what you mean. The self test in this case returns whatever error code is returned by `mbedtls_ripemd160_ext()`. Alternatively, it can return `1` in only one case: when the result does not match the expected value. I am not sure it is worth adding an error code just for this case. As you mentioned, this happens all over the library.